satpy/readers/ici_l1b_nc.py

```diff
@@ -0,0 +1,441 @@
+            'cached_solar_zenith',
+            'cached_solar_azimuth',
+        ] and getattr(self, var_key) is not None:
+            variable = getattr(self, var_key).copy()
+        else:
+            variable = self[var_key]
+        return variable
+
+    def get_dataset(self, dataset_id, dataset_info):
+        """Get dataset using file_key in dataset_info."""
+        var_key = dataset_info['file_key']
+        logger.debug(f'Reading in file to get dataset with key {var_key}.')
+        try:
+            variable = self._fetch_variable(var_key)
+        except KeyError:
+            logger.warning(f'Could not find key {var_key} in NetCDF file, no valid Dataset created')  # noqa: E501
+            return None
+        variable = self._filter_variable(variable, dataset_info)
+        # Perform the calibration if required
+        if dataset_info.get('calibration') is not None:
+            variable = self._perform_calibration(variable, dataset_info)
+        # Perform the orthorectification if required
+        if self.orthorect:
+            orthorect_data_name = dataset_info.get('orthorect_data', None)
+            if orthorect_data_name is not None:
+                variable = self._perform_orthorectification(
+                    variable,
+                    orthorect_data_name,
+                )
+        # Manage the attributes of the dataset
+        variable.attrs.setdefault('units', None)
+        variable.attrs.update(dataset_info)
+        variable.attrs.update(self._get_global_attributes())
+        variable = self._drop_coords(variable, 'n_horns')
+        variable = self._standardize_dims(variable)
+        return variable
+
+    def _get_global_attributes(self):
+        """Create a dictionary of global attributes to be added to all datasets."""
+        attributes = {
+            'filename': self.filename,
+            'start_time': self.start_time,
+            'end_time': self.end_time,
+            'spacecraft_name': self.spacecraft_name,
+            'ssp_lon': self.ssp_lon,
+            'sensor': self.sensor,
+            'filename_start_time': self.filename_info['sensing_start_time'],
+            'filename_end_time': self.filename_info['sensing_end_time'],
+            'platform_name': self.spacecraft_name,
+        }
+
+        # Add a "quality_group" item to the dictionary with all the variables
+        # and attributes which are found in the 'quality' group of the product
+        quality_group = self['quality']
+        quality_dict = {}
+        for key in quality_group:
+            # Add the values (as Numpy array) of each variable in the group
+            # where possible
+            try:
+                quality_dict[key] = quality_group[key].values
+            except ValueError:
+                quality_dict[key] = None
+        # Add the attributes of the quality group
+        quality_dict.update(quality_group.attrs)
+        attributes['quality_group'] = quality_dict
+        return attributes
```

Comment:
These comments indicate that the function could be made of smaller ones :)

Reply:
Agree :). Made smaller functions.

Reply:
thanks, very nice